### PR TITLE
ZIO.option do not catch throwable

### LINKED ISSUE #1868: ZIO.option shadows unchecked exceptions
```scala
@ r.unsafeRun(ZIO.dieMessage("oops").option)
res5: Option[Nothing] = None
```

In my opinion recovering from unchecked exceptions should be explicit and can be achieved by user with method `.sandbox`:
```scala
ZIO.dieMessage("oops").sandbox.option
```

### PR DESCRIPTION
**Pull request by darl**
fixes #1868

### CODE REVIEW

**Comment by iravid** [comment]
Thanks @darl. This makes sense to me. Let's wait for another reviewer's opinion. Could you resolve conflicts and run a format in the mean time?

File: core/shared/src/main/scala/zio/ZIO.scala [hunk 1]
```diff
@@ -646,7 +646,7 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
    * Executes this effect, skipping the error but returning optionally the success.
    */
   final def option: ZIO[R, Nothing, Option[A]] =
-    self.foldCauseM(_ => IO.succeed(None), a => IO.succeed(Some(a)))
+    self.foldM(_ => IO.succeed(None), a => IO.succeed(Some(a)))
```
Comment (jdegoes) [inline core/shared/src/main/scala/zio/ZIO.scala]: This looks good, can you make sure `either` is consistent with this?

Comment (adamgfraser) [reply core/shared/src/main/scala/zio/ZIO.scala]: Just checked and `either` is implemented in terms of `foldM` versus `foldCauseM`, so I think we are good here. Also reinforces that this is the right way to go as these methods should be consistent.

**Comment by jdegoes** [comment]
@darl Thank you for your work on this!

**Comment by darl** [comment]
👍 
p.s. looking for another free issue ;-)

**Comment by adamgfraser** [comment]
@darl I'm looking for help on #1898 if you are interested.

**Comment by darl** [comment]
@adamgfraser i'll look into it

**Pull request merged**